third_party/terraform/resources/resource_compute_router_nat.go

```diff
@@ -0,0 +1,392 @@
+package google
+
+import (
+	"fmt"
+	"log"
+
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	computeBeta "google.golang.org/api/compute/v0.beta"
+
+	"google.golang.org/api/googleapi"
+)
+
+var (
+	routerNatSubnetworkConfig = &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"source_ip_ranges_to_nat": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"secondary_ip_range_names": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+)
+
+func resourceComputeRouterNat() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceComputeRouterNatCreate,
+		Read:   resourceComputeRouterNatRead,
+		Delete: resourceComputeRouterNatDelete,
```

Comment:
I just did it this way at first. The documentation doesn't say it can't be updated, and you can in fact update it in Pantheon. Do you want me to implement that in this PR, or can I leave that to a follow up?

Reply:
A follow-up is totally fine. Maybe add a TODO?

Reply:
Done.